### PR TITLE
EncrypterOptions: Permit overriding the IV used for encryption

### PR DESCRIPTION
**Pull request by erincandescent**

### CODE REVIEW

File: symmetric_test.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func TestAeadErrors(t *testing.T) {
 		},
 	}
 
-	parts, err := aead.encrypt([]byte{}, []byte{}, []byte{})
+	parts, err := aead.encrypt([]byte{}, []byte{}, []byte{}, []byte{})
```
Comment (liclac) [inline symmetric_test.go]: Same here.

File: symmetric_test.go [hunk 2]
```diff
@@ -134,7 +134,7 @@ func TestVectorsAESGCM(t *testing.T) {
 
 	enc := newAESGCM(32)
 	key, _, _ := randomKeyGenerator{size: 32}.genKey()
-	out, err := enc.encrypt(key, aad, plaintext)
+	out, err := enc.encrypt(key, []byte{}, aad, plaintext)
```
Comment (liclac) [inline symmetric_test.go]: And here.

File: symmetric.go [hunk 3]
```diff
@@ -272,7 +276,7 @@ func (ctx *symmetricKeyCipher) encryptKey(cek []byte, alg KeyAlgorithm) (recipie
 	case A128GCMKW, A192GCMKW, A256GCMKW:
 		aead := newAESGCM(len(ctx.key))
 
-		parts, err := aead.encrypt(ctx.key, []byte{}, cek)
+		parts, err := aead.encrypt(ctx.key, nil, []byte{}, cek)
```
[the remark this answers is not in the document]
Comment (erincandescent) [reply symmetric.go]: This is correct; this is for non-direct modes; the random CEK (Content Encryption Key) is encrypted using AES+GCM with a random IV here; then later on we use the random CEK with the specified IV.

File: symmetric_test.go (hunk 1 above)
Comment (erincandescent) [reply symmetric_test.go]: This test overrode the RandReader used to read the IV above (Maybe we should change that but I wanted something minimally invasive)

File: symmetric_test.go (hunk 2 above)
Comment (erincandescent) [reply symmetric_test.go]: And same

**Pull request merged**